packages/cli/src/lib/project/templates/downloadProjectTemplate.ts

```diff
@@ -0,0 +1,68 @@
+import { UrlFormat } from "../../../commands";
+import { createUUID } from "../../helpers";
+import { runCommand } from "../../system";
+import { Logger } from "../../logging";
+
+import path from "path";
+import fse from "fs-extra";
+
+function parseUrlFormat(url: string): UrlFormat | undefined {
+  if (url.startsWith("http") && url.endsWith(".git")) {
+    return UrlFormat.git;
+  } else {
+    return undefined;
+  }
+}
+
+function createCacheDir(): string {
+  const cacheDir = `.polywrap/templates/${createUUID()}`;
+  if (!fse.existsSync(cacheDir)) {
+    fse.mkdirSync(cacheDir, { recursive: true });
+  }
+  return cacheDir;
+}
+
+function downloadGitTemplate(
+  url: string,
+  projectDir: string,
+  logger: Logger,
+  cacheDir: string
+): Promise<void> {
+  const command = "git clone";
+  const args = ["--depth", "1", "--single-branch", url];
+  const repoName = path.basename(url, ".git");
+  const repoDir = path.join(cacheDir, repoName);
+  const dotGitPath = path.join(repoDir, "/.git/");
+
+  return new Promise((resolve, reject) => {
+    // clone repo
+    runCommand(command, args, logger, undefined, cacheDir)
+      .catch(() => reject({ command: `${command} ${args.join(" ")}` }))
```

Comment:
i know this is still draft but it would be awesome if we use `await` and `try/catch` instead of `then/catch`

Reply:
I can do that, but why? I actually wrote it both ways and imo it looks cleaner this way. It's just a style preference, right?

Reply:
I refactored to use `await` with `try/catch`

Reply:
yes! its just a style preference - I think we should follow one pattern in the codebase hehe, just for the sake of consistency

Reply:
sounds good!